homeassistant/components/person/__init__.py

```diff
@@ -0,0 +1,168 @@
+"""
+Support for tracking people.
+
+For more details about this component, please refer to the documentation.
+https://home-assistant.io/components/person/
+"""
+import logging
+
+import voluptuous as vol
+
+from homeassistant.components.device_tracker import (
+    DOMAIN as DEVICE_TRACKER_DOMAIN)
+from homeassistant.const import ATTR_LATITUDE, ATTR_LONGITUDE
+from homeassistant.core import callback
+import homeassistant.helpers.config_validation as cv
+from homeassistant.helpers.entity_component import EntityComponent
+from homeassistant.helpers.event import async_track_state_change
+from homeassistant.helpers.restore_state import RestoreEntity
+
+_LOGGER = logging.getLogger(__name__)
+ATTR_FIRST_NAME = 'first_name'
+ATTR_LAST_NAME = 'last_name'
+ATTR_SOURCE = 'source'
+ATTR_USER_ID = 'user_id'
+CONF_DEVICE_TRACKERS = 'device_trackers'
+CONF_FIRST_NAME = 'first_name'
+CONF_LAST_NAME = 'last_name'
+CONF_USER_ID = 'user_id'
+DOMAIN = 'person'
+ENTITY_ID_FORMAT = DOMAIN + '.{}'
+
+PERSON_SCHEMA = vol.Schema({
+    vol.Required(CONF_FIRST_NAME): cv.string,
+    vol.Optional(CONF_LAST_NAME): cv.string,
+    vol.Optional(CONF_USER_ID): cv.string,
+    vol.Optional(CONF_DEVICE_TRACKERS, default=[]): vol.All(
+        cv.ensure_list, cv.entities_domain(DEVICE_TRACKER_DOMAIN)),
+})
+
+CONFIG_SCHEMA = vol.Schema({
+    DOMAIN: vol.All(cv.ensure_list, [PERSON_SCHEMA])
+}, extra=vol.ALLOW_EXTRA)
+
+
+async def async_setup(hass, config):
+    """Set up the Person component."""
+    component = EntityComponent(_LOGGER, DOMAIN, hass)
+    conf = config[DOMAIN]
+    entities = []
+    for person_conf in conf:
+        entities.append(Person(person_conf))
+
+    await component.async_add_entities(entities)
+
+    return True
+
+
+class Person(RestoreEntity):
+    """Represent a tracked Person."""
+
+    def __init__(self, config):
+        """Set up person."""
+        self._first_name = config[CONF_FIRST_NAME]
+        self._last_name = config.get(CONF_LAST_NAME)
+        self._latitude = None
+        self._longitude = None
+        self._source = None
+        self._state = None
+        self._trackers = config.get(CONF_DEVICE_TRACKERS)
+        self._user_id = config.get(CONF_USER_ID)
```

Comment:
We should validate that user_id is a valid user ID `await hass.auth.async_get_user(user_id) is not None`